### PR TITLE
add optional rkyv support

### LINKED ISSUE #583: rkyv Support
https://rkyv.org/

A zero-copy deserialization framework for rust. A sort of alternative to serde that has higher performance. chrono should add a feature gate where specifying "rkyv" will enable support for rkyv.

Types that should be serialized/deserialized should be given `#[derive(Archive, Serialize, Deserialize, PartialEq, Debug)]`.

### PR DESCRIPTION
**Pull request by dovahcrow**
### Thanks for contributing to chrono!

- [x] Have you added yourself and the change to the [changelog]? (Don't worry
      about adding the PR number)
- [ ] If this pull request fixes a bug, does it add a test that verifies that
      we can't reintroduce it?

[changelog]: ../CHANGELOG.md

Fixes #583

### CODE REVIEW

**Comment by djc** [comment]
This seems to have a lot of unrelated of changes. I'm afraid the changes in #661 have caused quite a few conflicts, but if you would rebase this on current main and make only the minimal changes necessary to support rkyv I'd be happy to review this.

**Comment by dovahcrow** [comment]
@djc Done.

File: src/date.rs [hunk 1]
```diff
@@ -18,7 +18,8 @@ use crate::naive::{self, IsoWeek, NaiveDate, NaiveTime};
 use crate::offset::{TimeZone, Utc};
 use crate::DateTime;
 use crate::{Datelike, Weekday};
-
+#[cfg(feature = "rkyv")]
+use rkyv::{Archive, Deserialize, Serialize};
```
Comment (djc) [inline src/date.rs:22]: Please put this in a separate block of imports, between the std/core block and the crate-internal imports. (Same for all the other imports.)

**Review by djc — COMMENTED** [review]
Thanks for rebasing!

File: src/date.rs (hunk 1 above)
Comment (djc) [reply src/date.rs:22]: It's not necessary for me to merge this PR, but if you don't mind doing the work that would be great! Please do it as a separate commit.

Comment (dovahcrow) [reply src/date.rs]: done

**Review by djc — APPROVED** [review]
LGTM, thanks!

**Pull request merged**